support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -41,45 +40,39 @@ class CreateZuoraSubscription(servicesProvider: ServiceProvider = ServiceProvide
     context: Context,
     services: Services
   ): FutureHandlerResult = {
-    val now = () => DateTime.now(DateTimeZone.UTC)
-    val today = () => LocalDate.now(DateTimeZone.UTC)
-    val promotionService = services.promotionService
-    val redemptionService = services.redemptionService
-    val zuoraGiftService = services.zuoraGiftService
-    val zuoraService = services.zuoraService
-    val giftCodeGenerator = services.giftCodeGenerator
-    val isTestUser = state.user.isTestUser
-    val config: ZuoraConfig = services.config.zuoraConfigProvider.get(isTestUser)
+
+    val createSubscription = {
+      val now = () => DateTime.now(DateTimeZone.UTC)
+      val isTestUser = state.user.isTestUser
+      val config: ZuoraConfig = services.config.zuoraConfigProvider.get(isTestUser)
+      new ZuoraSubscriptionCreator(now, services.promotionService, services.redemptionService, services.zuoraService, services.giftCodeGenerator, config)
+    }
 
     maybeDigitalSubscriptionGiftRedemption(state.product, state.paymentMethod) match {
       case Some(redemptionData) =>
-        redeemGift(redemptionData, requestInfo, state, zuoraGiftService, services.catalogService)
+        redeemGift(redemptionData, requestInfo, state, services.zuoraGiftService, services.catalogService)
       case None =>
-        createSubscription(state, requestInfo, config, Impure(now, today, promotionService, redemptionService, zuoraService, giftCodeGenerator))
+        createSubscription.create(state, requestInfo)
```

Comment:
it does seem much cleaner in terms of following the logic by getting rid of all the services code from this line.